src/MediaSource/MediaSourceService.php

```diff
@@ -289,8 +289,18 @@ public function putToNode(
       }
 
       $directory = $this->fileSystem->dirname($content_location);
+      
+      // Check if the directory is writable.
+      if (!is_writable(dirname($directory))) {
+        $parent_directory = dirname($directory);
+        $error_current_user = shell_exec('whoami');
+        $sanitized_current_user = str_replace(array("\n", "\r"), '', $error_current_user);
+        throw new HttpException(500, "The user '$sanitized_current_user' does not have permissions to write to: '$parent_directory' . Error");
+      }
```

Comment:
Additionally, if there's any chance of these messages being emitted to browsers, I'm not sure that it's particularly appropriate to include things like the names of OS-level users and the directory? Like... sure, if we want to write them to logs for the consumption of users privileged to read them, that's one thing, but displaying to general users, not so much?

Reply:
I think @adam-vessey is right; although I wouldn't object to the whoami bit being moved into the following if statement so that 500 error could read something like "The destination directory does not exist, could not be created, or is not writable by $current_user: $directory".

Reply:
Good point. Instead of adding this onto the thrown 500 exception, we could use `\Drupal::logger('islandora')->warning()` or whatever the dependency injection equivalent is.

Reply:
@adam-vessey I can move stuff into the existing one. I don't believe these are (I haven't seen them show up in the browser). I wasn't sure how much info is too much but right now it's not enough (in my opinion). Giving the user the directory is actionable. Telling them the username is probably fine but most people use usernames like nginx or www-data which is kinda standard so I wasn't sure how significant that would be for having in the logs.

Reply:
After thinking about it I do think the added value shouldn't be in a new if statement but included (if at all) in the existing one.

Reply:
Oddly enough, that error only really fires when derivatives generation is needing to create a new folder. It may just be how I'm testing this. But I thought it would be worth noting.